src/ObjectMultiplex.ts

```diff
@@ -0,0 +1,93 @@
+import { Duplex } from 'readable-stream';
+import eos from 'end-of-stream';
+import once from 'once';
+import { Substream } from './Substream';
+
+const IGNORE_SUBSTREAM: Record<string, unknown> = {};
+
+export class ObjectMultiplex extends Duplex {
+
+  private _substreams: Record<string, Substream | Record<string, any>>;
```

Comment:
Do we need to use a union type here? Just `Substream` won't work?

Reply:
I used the union because while using the type Record<string, Substream>, assigning an `empty record` `{}` as `IGNORE_STREAM` is throwing error. Is there an alternate approach to follow?

Reply:
Ah, I see! If you convert the `IGNORE_SUBSTREAM` to a `Symbol` like I suggest in another comment, you can change the `_substreams` type to `Record<string, Substream | IGNORE_SUBSTREAM>` (you possibly have to do `typeof IGNORE_SUBSTREAM`, I forget).